llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -76,11 +77,21 @@ def __init__(self, config: MPTConfig):
         config._validate_config()
         super().__init__(config)
 
-        self.attn_impl = config.attn_config['attn_impl']
-        self.prefix_lm = config.attn_config['prefix_lm']
-        self.attn_uses_sequence_id = config.attn_config['attn_uses_sequence_id']
-        self.alibi = config.attn_config['alibi']
-        self.alibi_bias_max = config.attn_config['alibi_bias_max']
+        # necessary to copy in order to be able to pop arguments so that we can pass in
+        # a subset of arguments into the Attention() init function as **kwargs
+        # and remove unnecessary args and still support checkpointing the original att_config
+        config.attn_config_mutable = copy.deepcopy(config.attn_config)
```

Comment:
This was a workaround to make `**attn_config` work, because certain arguments like `attn_type, prefix_lm, alibi` are not used in `MultiHeadAttention()`, but we want to be able to use `**attn_config`. Other possible workarounds:

```
1. .pop() when using these arguments (but I wasn't sure if this messes up checkpoints?)
2. **kwargs in the init arguments (don't love this as you know)
3. adding some of these arguments into the __init__ function of Attention
4. removing these arguments manually before passing into the init function ie.  attn_config_subset = {key: value for key, value in att_config.items() if key not in [attn_type, prefix_lm, alibi]}
```

Reply:
addressed